pkg/frontend/querymiddleware/remote_read.go

```diff
@@ -337,3 +340,43 @@ func cloneRemoteReadQuery(orig *prompb.Query) (*prompb.Query, error) {
 
 	return cloned, nil
 }
+
+type remoteReadSanitizerMiddleware struct {
+	next   MetricsQueryHandler
+	logger log.Logger
+}
+
+func newRemoteReadSanitizerMiddleware(logger log.Logger) MetricsQueryMiddleware {
+	return MetricsQueryMiddlewareFunc(func(next MetricsQueryHandler) MetricsQueryHandler {
+		return &remoteReadSanitizerMiddleware{
+			next:   next,
+			logger: logger,
+		}
+	})
+}
+
+func (r *remoteReadSanitizerMiddleware) Do(ctx context.Context, req MetricsQueryRequest) (Response, error) {
```

Comment:
I'm not convinced about the need of the sanitizer. Can't we simply always return 0 from `remoteReadQueryRequest.GetStep()` adding a comment to explain why?

Reply:
That was my first reaction as well, but I was afraid of the ingester ending up doing something else.  I'll revert this change and see if I can add a unit test that defines the ingester behavior.